### ISSUE TITLE
Implement new classical optimizers in qiskit 

### ISSUE THREAD

**Issue by BryceFuller** [qiskit-community/qiskit-advocate-mentorship-program#22]
# Description

New optimization algorithms are being developed in the context of classical machine learning, which could find application in training quantum circuit parameters. The core of this project will be to work with your mentor to identify, implement, and document 2-3 promising classical optimizers for use in qiskit. For example, an optimizer might be considered 'promising' by offering noise robustness while requiring comparatively few queries to quantum hardware. 

I already have one optimizer in mind, known as [Sharpness Aware Minimization (SAM)](https://arxiv.org/pdf/2010.01412v2.pdf). This optimizer seeks to find model parameters which are robust to small perturbations. When executing parameterized rotations on quantum hardware, our gates have some amount of error. Thus SAM could prove useful.

# Mentor/s
Bryce Fuller (@BryceFuller), Quantum Computing Application Researcher

# Type of participant
Should be familiar with qiskit and feel comfortable with the concepts behind gradient descent. 
Experience with machine learning / optimization will prove useful. 

# Number of participants
1-2 depending on experience. 

# Deliverable
A PR into qiskit aqua containing the new optimizers, as well as a tutorial notebook explaining how to use them. 
If appropriate, participants may be encouraged to write a Medium blog post explaining the ideas behind the new optimizers.

**Comment by HuangJunye** [comment]
@molar-volume Can you comment on this issue so that I can assign you? Please also work with your mentor to refine the project, define scope and deliverables and update the project description in this issue.

**Comment by molar-volume** [comment]
Hi @HuangJunye,
sure you can assign it to me, thanks.
